qiskit/primitives/backend_estimator_v2.py

```diff
@@ -161,8 +161,11 @@ def _validate_pubs(self, pubs: list[EstimatorPub]):
     def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
         pub_dict = defaultdict(list)
         # consolidate pubs with the same number of shots
+        coeff_sq_sum = 0
```

Comment:
I am a little bit confused by the math here..

The variance of a linear combination `aX + bY + ...` of observables `X, Y, ...` is given by:

> `var(aX + bY + ...) = a**2 Var(X) + b**2 Var(Y) + ...`, 


where `Var(O) = 1/N_O` is the variance of the observable `O` and `N_O` is the number of shots for the observable `O`.
In our case, we want `var(aX + bY + ...)` to be equal to `epsilon**2`, and we want to use the same number of shots `N` for every observable `O`.  Therefore, we have: 

> `epsilon**2 = a**2/N + b**2/N + ...`, 


which gives:

> `N = (a**2 + b**2 + ...)/epsilon**2`

So here is how I would code this up:
```
for i, pub in enumerate(pubs):
    coeff_sq_sum = 0
    for obs in pub.observables.ravel():
        coeff_sq_sum += sum(coeff**2 for coeff in obs.values())
    shots = int(math.ceil(coeff_sq_sum / pub.precision**2))
```

Am I missing something?

Reply:
FYI, there's a related issue that may shed some light
https://github.com/Qiskit/qiskit-ibm-runtime/issues/1684

Reply:
Right. This is how it should be as written by @ihincks in https://github.com/Qiskit/qiskit/pull/12618. Given that we are estimating the precision without knowing the target state, setting Var(O)=1/N_0 bounds the shot number.

Reply:
> Am I missing something?
I think you might be: each element of `pub.observables.ravel()` it itself a multi-term observable. The point of the max was to find the worst offender.

Reply:
I see. So every `obs` is not an observable, but a list of (potentially more than one) observables.
@jsaroni could you add a comment about this? It will help us remember in the future